hr_employee_benefit/models/hr_employee_benefit_rate.py

```diff
@@ -100,22 +100,24 @@ def get_all_amount_types(self):
             ('annual', _('Annual')),
         ]
 
+    @api.multi
     def _get_amounts_now(self):
         today = context_today(self)
-        self.employee_amount = self.get_amount(today)
-        self.employer_amount = self.get_amount(today, employer=True)
-
+        for rate in self:
+            rate.employee_amount = rate.get_amount(today)
+            rate.employer_amount = rate.get_amount(today, employer=True)
+    
     @api.multi
     def get_amount(self, date, employer=False):
-        self.ensure_one()
```

Comment:
I think with the change of this pr in the method `_get_amounts_now` is enough to fix the error.
Because the problem is we had a:
```bash
@api.multi
def _get_amounts_now(self):
      self.employee_amount = self.get_amount(today)  #  <-  Error here using a singleton method with a non-singleton `self`
```

With your fix 
```bash
@api.multi
def _get_amounts_now(self):
     for record in self:
          record.employee_amount = self.get_amount(today)  #  <-  Good. Using a singleton method with a singleton `record`
```